### PR TITLE
varnish 7.0.1

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
varnish:
  * Libraries were compiled with a flat namespace.
    This can cause linker errors due to name collisions, and
    is often due to a bug in detecting the macOS version.
      /opt/homebrew/Cellar/varnish/7.0.0_1/lib/libvarnishapi.3.dylib
```

**Review by branchv — DISMISSED** [review]

**Comment by carlocab** [comment]
@chenrui333 FYI:
```
❯ brew unbottled --tag=arm64_monterey varnish
==> Populating dependency tree...
==> :arm64_monterey bottle status
varnish: unbottled deps: apr apr-util subversion gts librsvg graphviz
❯ brew unbottled --tag=monterey varnish
==> Populating dependency tree...
==> :monterey bottle status
varnish: unbottled deps: apr apr-util subversion gts graphviz
```

**Comment by cclauss** [comment]
What is the remaining TODO on this PR now that ~#88019~ and ~#88018~ have been merged?

**Comment by carlocab** [comment]
Fix the test failure on Catalina. Let's run it again to double-check.

**Comment by cclauss** [comment]
Perhaps Varnish 7.0.1 instead of 7.0.0?
https://github.com/varnishcache/varnish-cache/blob/7.0/doc/changes.rst#varnish-cache-701-2021-11-23

File: Formula/varnish.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class Varnish < Formula
   mirror "https://fossies.org/linux/www/varnish-7.0.0.tgz"
   sha256 "8c7a5c0b1f36bc70bcbc9a48830835249e895fb8951f0363110952148cbae087"
```
Comment (cclauss) [inline Formula/varnish.rb:6]: https://varnish-cache.org/releases/
## Line 4 would also need to be updated to v7.0.1
```suggestion
  mirror "https://fossies.org/linux/www/varnish-7.0.1.tgz"
  sha256 "c4e75beff0d461611742361fe8039ee1233ddf755b2b8a1e18a5fcacbe2b4660"
```

File: Formula/varnish.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Varnish < Formula
...
   license "BSD-2-Clause"
-  revision 1
+  revision 2
```
Comment (cclauss) [inline Formula/varnish.rb:8]:
```suggestion
  revision 1
```

**Review by cclauss — COMMENTED** [review]
Please try upgrading to Varnish v7.0.1.  https://varnish-cache.org/releases/

**Comment by carlocab** [comment]
Some of these tests seem to be randomly failing though. In the last run, Big Sur was fine and Catalina errored. Now it's the reverse. I wonder if we should be running fewer of these tests. CC @mjpieters

File: Formula/varnish.rb (hunk 2 above)
Comment (carlocab) [reply Formula/varnish.rb:8]: This line should be removed for version bumps, actually.

**Comment by cclauss** [comment]
`/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/lib/ruby/2.6.0/timeout.rb:108:in `timeout'`

Is there a way to increase the timeout?

**Comment by carlocab** [comment]
> `/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/lib/ruby/2.6.0/timeout.rb:108:in 'timeout'`
> 
> Is there a way to increase the timeout?

No. The timeout looks spurious, though.

**Review by carlocab — APPROVED** [review]
Ok, that worked. But if the tests start timing out again we should probably start cutting them down.

File: Formula/varnish.rb [hunk 3]
```diff
@@ -22,7 +21,7 @@ class Varnish < Formula
   depends_on "docutils" => :build
   depends_on "graphviz" => :build
   depends_on "pkg-config" => :build
-  depends_on "python@3.9" => :build
+  depends_on "python@3.10" => :build
```
Comment (mjpieters) [inline Formula/varnish.rb:24]: I don't see an update for the Python version in `install`:

```ruby

ENV["PYTHON"] = Formula["python@3.9"].opt_bin/"python3"

```

That'll need updating too.

Comment (carlocab) [reply Formula/varnish.rb:24]: #89929

**Pull request closed without merging**